On empty tmp/* I check this after eating :

FAILED unit/test_router.py::test_router_models_max_evicts_lru - AssertionError: Timed out waiting for ggml-org/tinygemma3-GGUF:Q8_0 to reach {'loaded'}, last status: unloaded

--- a/tools/server/tests/unit/test_basic.py
+++ b/tools/server/tests/unit/test_basic.py
@@ -65,6 +65,7 @@ def test_server_slots():
 
 def test_load_split_model():
     global server
+    server.offline = False
     server.model_hf_repo = "ggml-org/models"
     server.model_hf_file = "tinyllamas/split/stories15M-q8_0-00001-of-00003.gguf"
     server.model_alias = "tinyllama-split"

--- a/tools/server/tests/unit/test_router.py
+++ b/tools/server/tests/unit/test_router.py
@@ -17,7 +17,6 @@ def create_server():
     ]
 )
 def test_router_chat_completion_stream(model: str, success: bool):
-    # TODO: make sure the model is in cache (ie. ServerProcess.load_all()) before starting the router server
     global server
     server.start()
     content = ""
@@ -108,26 +107,22 @@ def test_router_models_max_evicts_lru():
         "ggml-org/test-model-stories260K-infill",
     ]
 
-    loaded_models: list[str] = []
-    for model_id in candidate_models:
-        try:
-            _load_model_and_wait(model_id, timeout=120)
-            loaded_models.append(model_id)
-        except AssertionError:
-            continue
+    # Load only the first 2 models to fill the cache
+    first, second, third = candidate_models[:3]
 
-    if len(loaded_models) < 3:
-        pytest.skip("Not enough models could be loaded to exercise eviction")
+    _load_model_and_wait(first, timeout=120)
+    _load_model_and_wait(second, timeout=120)
 
-    first, second, third = loaded_models[:3]
+    # Verify both models are loaded
+    assert _get_model_status(first) == "loaded"
+    assert _get_model_status(second) == "loaded"
 
-    _wait_for_model_status(first, {"loaded"})
-    _wait_for_model_status(second, {"loaded"})
-
+    # Load the third model - this should trigger LRU eviction of the first model
     _load_model_and_wait(third, timeout=120)
 
+    # Verify eviction: third is loaded, first was evicted
     assert _get_model_status(third) == "loaded"
-    assert _get_model_status(first) != "loaded"
+    assert _get_model_status(first) == "unloaded"
 
 
 def test_router_no_models_autoload():

--- a/tools/server/tests/utils.py
+++ b/tools/server/tests/utils.py
@@ -554,6 +554,7 @@ class ServerPreset:
     @staticmethod
     def router() -> ServerProcess:
         server = ServerProcess()
+        server.offline = True # will be downloaded by load_all()
         # router server has no models
         server.model_file = None
         server.model_alias = None

--- a/tools/server/tests/utils.py
+++ b/tools/server/tests/utils.py
@@ -450,6 +450,7 @@ class ServerPreset:
     @staticmethod
     def tinyllama2() -> ServerProcess:
         server = ServerProcess()
+        server.offline = True # will be downloaded by load_all()
         server.model_hf_repo = "ggml-org/test-model-stories260K"
         server.model_hf_file = None
         server.model_alias = "tinyllama-2"

